client/packages/invoices/src/Returns/api/hooks/document/useOutboundDeleteRows.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  useTableStore,
+  useTranslation,
+  useQueryClient,
+  InvoiceNodeStatus,
+  useDeleteConfirmation,
+  useUrlQueryParams,
+  useMutation,
+} from '@openmsupply-client/common';
+import { useOutbounds } from './useOutbounds';
+import { useReturnsApi } from '../utils/useReturnsApi';
+
+export const useOutboundDeleteRows = () => {
+  const queryClient = useQueryClient();
+  const { queryParams } = useUrlQueryParams({
+    initialSort: { key: 'createdDatetime', dir: 'desc' },
+  });
+  const { data: rows } = useOutbounds(queryParams);
+  const api = useReturnsApi();
+  const { mutateAsync } = useMutation(api.deleteOutbound);
+  const t = useTranslation('replenishment');
+
+  const selectedRows = useTableStore(
+    state =>
+      rows?.nodes.filter(({ id }) => state.rowState[id]?.isSelected) ?? []
+  );
+
+  const deleteAction = async () => {
+    await mutateAsync(selectedRows)
+      .then(() => queryClient.invalidateQueries(api.keys.base()))
+      .catch(err => {
+        throw err;
+      });
+  };
+
+  const confirmAndDelete = useDeleteConfirmation({
+    selectedRows,
+    deleteAction,
+    canDelete: selectedRows.every(
+      ({ status }) => status === InvoiceNodeStatus.New // is this accurate?
```

Comment:
The outbound shipment uses  `canDeleteInvoice` function -- can we use the same one?

Looks like the criteria is  `invoice.status === InvoiceNodeStatus.New ||  invoice.status === InvoiceNodeStatus.Allocated`, but I'm not sure if that's exactly what we'll have. We haven't really discussed statuses yet. Fine to leave a "TO DO" here I guess?